dotCMS/src/main/java/com/dotcms/rest/api/v1/folder/FolderHelper.java

```diff
@@ -25,93 +24,58 @@
             new TreeableNameComparator();
 
     private FolderHelper() {
         this.hostAPI = APILocator.getHostAPI();
         this.folderAPI = APILocator.getFolderAPI();
-        this.userAPI = APILocator.getUserAPI();
     }
 
     private static class SingletonHolder {
         private static final FolderHelper INSTANCE = new FolderHelper();
     }
 
-    protected class FolderResults {
-
-        protected FolderResults(List<Folder> folders, List<Map<String, String>> errors) {
-            this.setErrors(errors);
-            this.setFolders(folders);
-        }
-
-        List<Folder> folders;
-        /**
-         * Map of any errors found while saving a folder.  The map has 2 attributes. failedPath and errorMessage
-         */
-        List<Map<String, String>> errors;
-
-        protected List<String> getErrorKeys() {
-            List<String> keys = new ArrayList<String>();
-            keys.add("failedPath");
-            keys.add("errorMessage");
-            return keys;
-        }
-
-        protected List<Folder> getFolders() {
-            return folders
-                    .stream()
-                    .collect(Collectors.toList());
-        }
-
-        protected void setFolders(List<Folder> folders) {
-            this.folders = folders;
-        }
-
-        protected List<ErrorEntity> getErrorEntities() {
-            List<ErrorEntity> ret = new ArrayList<ErrorEntity>();
-            for (Map<String,String> error : errors) {
-                ErrorEntity ee = new ErrorEntity(error.get("failedPath"),error.get("errorMessage"));
-                ret.add(ee);
-            }
-            return ret;
-        }
-
-        protected void setErrors(List<Map<String, String>> errors) {
-            this.errors = errors;
-        }
-    }
-
     /**
      * Get the instance.
      *
-     * @return BrowserTreeHelper
+     * @return FolderHelper
      */
     public static FolderHelper getInstance() {
 
         return FolderHelper.SingletonHolder.INSTANCE;
     } // getInstance.
 
-    public FolderResults createFolders(List<String> paths, String siteName, User user) throws DotDataException, DotSecurityException {
+    /**
+     * Creates all folders and subfolders passed in the list.
+     *
+     * @param paths List of folders to create
+     * @param siteName siteName where the folders are gonna be created
+     * @param user user to create folders
+     * @return list of folders created
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    public List createFolders(final List<String> paths, final String siteName, final User user) throws DotDataException, DotSecurityException {
 
-        List<Folder> savedFolders = new ArrayList<Folder>();
-        List<Map<String, String>> errors = new ArrayList<Map<String, String>>();
-        Host host = hostAPI.findByName(siteName, user, true);
+        final List<Folder> savedFolders = new ArrayList<Folder>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5552986)